torch/testing/_internal/distributed/nn/api/remote_module_test.py

```diff
@@ -0,0 +1,130 @@
+import enum
+from typing import Tuple
+
+import torch
+import torch.testing._internal.dist_utils as dist_utils
+from torch import Tensor, nn
+from torch.distributed.nn.api.remote_module import RemoteModule, _gen_global_unique_name
+from torch.testing._internal.distributed.rpc.rpc_agent_test_fixture import (
+    RpcAgentTestFixture,
+)
+
+
+class ModuleCreationMode(enum.Enum):
+    SCRIPT_MODULE_CREATOR = "script_module_creator"
+    MODULE_CTOR_WITH_INTERFACE = "module_ctor_with_interface"
+    MODULE_CTOR = "module_ctor"
+
+
+@torch.jit.interface
+class MyModuleInterface:
+    def forward(
+        self, tensor: Tensor, number: int, word: str = "default"
+    ) -> Tuple[str, int, Tensor]:
+        pass
+
+
+class MyModule(nn.Module):
```

Comment:
Can we verify something like `class MyScriptModule(torch.jit.ScriptModule):` works too?

Reply:
That's a deprecated API which is not used by us internally. I don't think we want to support that.